include/knowhere/clustering/clustering.h

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#ifndef CLUSTERING_H
+#define CLUSTERING_H
+
+#include "knowhere/binaryset.h"
+#include "knowhere/clustering/clustering_node.h"
```

Comment:
It is a little bit weird to make `clustering` = `index`. A better name is needed, some thing like `cluster` and `cluster_operator`. These just an example for reference.

Reply:
Lets go with cluster